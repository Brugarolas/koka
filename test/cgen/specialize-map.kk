fun map-int(xs : list<int>, f : int -> int) : list<int> {
  match (xs) {
    Cons(x, xx) -> Cons(x.f, xx.map-int(f))
    Nil -> Nil
  }
}

public fun test() {
  [1,2,3].map-int(fn(x){ x + 1 })
}

public fun main() {
  test().show.println
}
