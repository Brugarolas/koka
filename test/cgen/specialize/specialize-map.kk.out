cgen/specialize/specialize-map/.lift119-test: (xs : list<int>) -> list<int>
cgen/specialize/specialize-map/.lift120-main: (xs : list<int>) -> list<int>
cgen/specialize/specialize-map/main: () -> console ()
cgen/specialize/specialize-map/map-int: (xs : list<int>, f : (int) -> int) -> list<int>
cgen/specialize/specialize-map/test: () -> list<int>
